engines/scc_proxy/lib/scc_proxy/engine.rb

```diff
@@ -329,6 +369,17 @@ def scc_activate_product
             InstanceVerification.update_cache(request.remote_ip, @system.login, @product.id)
           end
         end
+
+        def de_register_hybrid(auth)
+          response = SccProxy.deregister_system_scc(auth, @system.system_token)
+          unless response.code_type == Net::HTTPNoContent
+            error = JSON.parse(response.body)
+            logger.info "Could not de-activate system #{@system.login}, error: #{error['error']} #{response.code}"
+            error['error'] = SccProxy.parse_error(error['error'], params[:token], params[:email]) if error['error'].include? 'json'
+            raise ActionController::TranslatedError.new(error['error'])
+          end
+          logger.info 'System successfully deregistered from SCC'
+        end
```

Comment:
Why do we need a separate implementation? I would have expected that we simply check if a system that is getting removed is hybrid and if that is the case call the existing `deregister_system_scc()` method. What is special about the hybrid system that requires this extra wrapper?

Reply:
> Why do we need a separate implementation? 

To not get `Avoid more than 3 levels of block nesting.` 
If the code inside ~de_register_hybrid~ `deregister_hybrid` method goes where that method is right now,
we would get  that message and rubocop would not be happy, making the linter fail

> if a system that is getting removed is hybrid

No system is getting removed on that call, a product of a system is getting **activated**, the condition is
if the system - whose product is getting activated - was just announced to SCC because it is `hybrid` scenario

>  if that is the case call the existing deregister_system_scc() method

Yes, but we need to handle the error, if any

> What is special about the hybrid system that requires this extra wrapper ?

Nothing special, just to avoid rubocop errors

We could move the error handling inside `SccProxy.deregister_system_scc`  which, if no one objects, I will do